### ISSUE TITLE
Will there be 10.0 update? Or the addon is dead?

### ISSUE THREAD

**Issue by aristes** [AdiAddons/AdiButtonAuras#404]
The game seems unplayable for me without it :(

**Comment by gelid** [comment]
temp fix is
in file \_retail_\Interface\AddOns\AdiButtonAurasAdiButtonAuras.toc
comment 
#SMBH\lib.xml

in file 
\_retail_\Interface\AddOns\AdiButtonAuras\core\Core.lua
comment
--hooksecurefunc('PetActionBar_Update', function()
	--	for i = 1, NUM_PET_ACTION_SLOTS do
	--		UpdateHandler('PetActionBar_Update', _G['PetActionButton'..i])
	--	end
	--end)
	--hooksecurefunc('StanceBar_UpdateState', function()
	--	for i = 1, NUM_STANCE_SLOTS do
	--		UpdateHandler('StanceBar_UpdateState', _G['StanceButton'..i])
	--	end
	--end)

then go https://www.curseforge.com/wow/addons/libbuttonglow-1-0/download/4040423
download lib
unpuk 
and replace same name libs in
\_retail_\Interface\AddOns\AdiButtonAuras\libs
LibButtonGlow-1.0
LibStub

settings dont work, but glow work for me

**Comment by mario-64** [comment]
Made those exact changes but not working. No glow and many LUA errors.

**Comment by gelid** [comment]
> Made those exact changes but it's not working. No glow and many LUA errors.

idk - why
im repeat all steps on fresh install addon and it work for me
may be some you own rules broke or other addons conflict

for replacing addon
in download lib there is 2 files and folder
files goes in lib'LibButtonGlow-1.0
but folder goes in lib'LibStub

**Comment by mario-64** [comment]
I suspect these changes are tainting other addons. The LUA errors were coming from them.

**Comment by shenjiayee** [comment]
> settings dont work, but glow work for me

there are something changed in `AceConfigDialog:AddToBlizOptions` for 10.0.
so, maybe we can update the lib-ace3 to solve the problem.

**Comment by aristes** [comment]
IT WORKS
You just need to open the folder from here
https://www.curseforge.com/wow/addons/libbuttonglow-1-0/download/4040423
and replace folder LibStub here \_retail_\Interface\AddOns\AdiButtonAuras\LibStub
and then 2 files "LibButtonGlow-1.0.lua" and "LibButtonGlow-1.0.lua" replace here \_retail_\Interface\AddOns\AdiButtonAuras\LibButtonGlow-1.0

**Comment by asrandor** [comment]
So the temp fix works, I do not have errors enymore but AdiButtonAura gives me this warning nonetheless :
![image](https://user-images.githubusercontent.com/29277120/198574115-beca70e2-9649-4c5f-a2d4-46f850b8df1d.png)

**Comment by shenjiayee** [comment]
> 

and you also need to fix `AdiButtonAuras/SMBH/SMBH.lua` - line 37,

```lua
local texturePath = "Interface\\AddOns\\AdiButtonAuras\\SMBH\\media"
```
just use a static path instead

besides, i got a tempfix for the settings.

1.  download the new ace3-config-lib to replace the lib files in the folder `\AdiButtonAuras\libs\AceConfig-3.0`- replace them with the same-name folder.
2.  in src file `AdiButtonAuras/libs/AceConfig-3.0/AceConfigDialog-3.0/AceConfigDialog-3.0.lua` - after line 2002, add code like this: ```

**Comment by tholanis** [comment]
> ```lua
> local texturePath = "Interface\\AddOns\\AdiButtonAuras\\SMBH\\media"
> ```
I tested  this and found out that textures could not loaded due to missing \ at path end
This fixed it:
`local texturePath = "Interface\\AddOns\\AdiButtonAuras\\SMBH\\media\\"`

Now I have the colored button borders back.

**Comment by reed-kevin** [comment]
> 

Any chance you can upload your modded code / zip somewhere, please?

**Comment by mario-64** [comment]
I’ve made these exact changes and still it doesn’t work. No glow and taints to other addons 😞

**Comment by reed-kevin** [comment]
Managed to get it working.  Uploaded [here](https://drive.google.com/file/d/1Dk-TgLgqtPSQ1UfAog28dhotudal5dvK/view?usp=share_link)

**Comment by persath** [comment]
Ok with so many fixes spread all over it might be hard to figure it out. But after 2 days of testing here goes.

From gelid original fix, what to do:

*** DO NOT DO THIS ***
in file
\_retail_\Interface\AddOns\AdiButtonAuras\AdiButtonAuras.toc
comment
#SMBH\lib.xml
*** DO NOT DO THIS ***

Step 1 ***

in file \_retail_\Interface\AddOns\AdiButtonAuras\core\Core.lua

comment by adding -- at the start of the line like bellow

--hooksecurefunc('PetActionBar_Update', function()
--	for i = 1, NUM_PET_ACTION_SLOTS do
--		UpdateHandler('PetActionBar_Update', _G['PetActionButton'..i])
--	end
--end)
--hooksecurefunc('StanceBar_UpdateState', function()
--	for i = 1, NUM_STANCE_SLOTS do
--		UpdateHandler('StanceBar_UpdateState', _G['StanceButton'..i])
--	end
--end)

Step 2 ***

Download https://www.curseforge.com/wow/addons/libbuttonglow-1-0/download/4040423
unzip
copy files LibButtonGlow-1.0.lua and LibButtonGlow-1.0.toc into _retail_\Interface\AddOns\AdiButtonAuras\libs\LibButtonGlow-1.0 and replace existing files
copy files LitStub\LibStub.lua and LitStub\LibStub.toc into _retail_\Interface\AddOns\AdiButtonAuras\libs\LibStub and replace existing files

From shenjiayee fix, what to do:

Step 3 ***

Edit AdiButtonAuras/SMBH/SMBH.lua - line 37,

local texturePath = "Interface\\AddOns\\AdiButtonAuras\\SMBH\\media\\"

this makes *** DO NOT DO THIS *** part not needed and also added the \\ at the end of the path like tholanis pointed out.
Also i tried the setting fix and it broke everything so try at your own risk.

All merit goes to gelid, shenjiayee and tholanis; i just put it all together in one post.

Final note some glows dont work due to changes in talents and spells, it's not the addon or fixes fault, guess we will have to wait for updated libs for that.

**Comment by mario-64** [comment]
Ok which is it?

local texturePath = "Interface\\AddOns\\AdiButtonAuras\\SMBH\\media\\"
OR
local texturePath = "Interface\AddOns\AdiButtonAuras\SMBH\media\"

One slash or two?

**Comment by persath** [comment]
> Ok which is it?
> 
> local texturePath = "Interface\AddOns\AdiButtonAuras\SMBH\media\" OR local texturePath = "Interface\AddOns\AdiButtonAuras\SMBH\media"
> 
> One slash or two?

two slashes in all, just updated the post sorry about that

**Comment by mario-64** [comment]
Thank you @persath ! Now it's working for me. What was breaking it evidently was the instruction to comment out #SMBH\lib.xml. As you mentioned that part is wrong.

**Comment by Zifijo** [comment]
Thank you @persath !!!! Auras are working now. I still get some errors when i log on or reload but so far so good xD

**Comment by shenjiayee** [comment]
@reed-kevin @persath @mario-64
I upload the fixed one to my space.
[https://github.com/shenjiayee/AdiButtonAuras/](url)
I will try to fixed the problem until the original author updates.

**Comment by Nulgar** [comment]
@shenjiayee thanks, but while the displayed URL is fine, I think the link is a bit broken ;)

**Comment by gelid** [comment]
found alternative 
https://www.curseforge.com/wow/addons/litebuttonauras
its work for me

**Comment by shenjiayee** [comment]
> Thank you @persath ! Now it's working for me. What was breaking it evidently was the instruction to comment out #SMBH\lib.xml. As you mentioned that part is wrong.

fixed

**Comment by shenjiayee** [comment]
> @shenjiayee thanks, but while the displayed URL is fine, I think the link is a bit broken ;)

@Nulgar 
fixed

**Comment by shenjiayee** [comment]
> found alternative https://www.curseforge.com/wow/addons/litebuttonauras its work for me

it is a lighter one, but it does not support flash on missing (de)buff.

**Comment by sthdevilish** [comment]
I tried the fix above, but it only produced more errors.

**Comment by reed-kevin** [comment]
`<Ui xmlns="http://www.blizzard.com/wow/ui/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.blizzard.com/wow/ui/
..\FrameXML\UI.xsd">
	<Script file="Utils.lua"/>
	<Script file="Core.lua"/>
	<Script file="SafeEnv.lua"/>
	<Script file="AuraTools.lua"/>
	<Script file="Description.lua"/>
	<Script file="RuleDSL.lua"/>
	<Script file="Items.lua"/>
	<Script file="Overlays.lua"/>
	<Script file="Display.lua"/>
</Ui>`

**Comment by reed-kevin** [comment]
Uploaded the file here: https://drive.google.com/file/d/18X_Lh7UVORK5tSs0Ij_-3Hgqj2qrQ2_x/view?usp=sharing

**Comment by sthdevilish** [comment]
Thanks reed-kevin! It works, I must have made a typo somewhere.

**Comment by SharonS420** [comment]
This addon is dead.  I wouldn't recommend any of you changes codes from people you don't know.  They aren't working anyhow.

**Comment by mario-64** [comment]
I agree. This addon is dead. I’ve switched to Lite Button Auras which is supported and working well for me.
